globals/matchmaking_utils.go

```diff
@@ -131,21 +128,171 @@ func RemoveConnectionFromAllSessions(connection *nex.PRUDPConnection) {
 			messagePacket.SetType(constants.DataPacket)
 			messagePacket.AddFlag(constants.PacketFlagNeedsAck)
 			messagePacket.AddFlag(constants.PacketFlagReliable)
-			messagePacket.SetSourceVirtualPortStreamType(connection.StreamType)
+			messagePacket.SetSourceVirtualPortStreamType(target.StreamType)
 			messagePacket.SetSourceVirtualPortStreamID(endpoint.StreamID)
-			messagePacket.SetDestinationVirtualPortStreamType(connection.StreamType)
-			messagePacket.SetDestinationVirtualPortStreamID(connection.StreamID)
+			messagePacket.SetDestinationVirtualPortStreamType(target.StreamType)
+			messagePacket.SetDestinationVirtualPortStreamID(target.StreamID)
 			messagePacket.SetPayload(rmcRequestBytes)
 
 			server.Send(messagePacket)
+
+			return false
+		})
+	}
+	
+	if SessionManagementDebugLog {
+		globals.Logger.Infof("GID %d: Deleted", gathering)
+	}
+	delete(sessions, gathering)
+}
+
+func RemoveSession(connection *nex.PRUDPConnection, gathering uint32) {
+	sessionsMutex.Lock()
+	defer sessionsMutex.Unlock()
+
+	removeSessionImpl(connection, gathering)
+}
+
+// RemoveConnectionIDFromSession removes a PRUDP connection from the session
+func removeConnectionIDFromSessionImpl(connection *nex.PRUDPConnection, gathering uint32, gracefully bool) {
+	session, ok := sessions[gathering]
+	if !ok {
+		return
+	}
+	session.ConnectionIDs.DeleteAll(connection.ID)
+
+	ownerPID := session.GameMatchmakeSession.Gathering.OwnerPID
+	lenParticipants := session.ConnectionIDs.Size()
+
+	if SessionManagementDebugLog {
+		var grace string
+		if gracefully {
+			grace = "gracefully"
+		} else {
+			grace = "ungracefully"
 		}
+		globals.Logger.Infof("GID %d: Removed PID %d %s", gathering, connection.PID().LegacyValue(), grace)
+	}
 
-		gid = FindConnectionSession(connection.ID)
+	// If there are no more participants remove the session
```

Comment:
Nitpick. This goes for all comments. We use the syntax from [Better Comments](https://marketplace.visualstudio.com/items?itemName=aaron-bond.better-comments) for comments, with the exception of `TODO` which is structured as `TODO -` rather than `TODO:`

```suggestion
	// * If there are no more participants remove the session
```